Libraries/plist/Tools/PlistBuddy.cpp

```diff
@@ -0,0 +1,572 @@
+ProcessCommand(libutil::Filesystem *filesystem, std::string path, bool xml, std::string const &file, RootObjectContainer &root, std::string const &input)
+{
+    std::vector<std::string> tokens;
+    std::stringstream sstream(input);
+    std::copy(std::istream_iterator<std::string>(sstream), std::istream_iterator<std::string>(), std::back_inserter(tokens));
+
+    if (tokens.size() < 1) {
+        return true;
+    }
+
+    std::string command = tokens[0];
+    if (command == "Print") {
+        std::queue<std::string> keyPath;
+        if (tokens.size() < 2) {
+            parseCommandKeyPathString("", keyPath);
+        } else {
+            parseCommandKeyPathString(tokens[1], keyPath);
+        }
+        Print(root.object.get(), keyPath, xml);
+    } else if (command == "Save") {
+        std::queue<std::string> keyPath;
+        parseCommandKeyPathString("", keyPath);
+        Print(root.object.get(), keyPath, xml, filesystem, path);
+    } else if (command == "Exit") {
+        return false;
+    } else if (command == "Set") {
+        if (tokens.size() < 2) {
+            std::cerr << "Set command requires KeyPath" << std::endl;
+            return true;
+        } else {
+            std::queue<std::string> keyPath;
+            parseCommandKeyPathString(tokens[1], keyPath);
+            Set(root.object.get(), keyPath, plist::ObjectType::String, parseCommandValueString(tokens.begin() + 2, tokens.end()));
+        }
+    } else if (command == "Add") {
+        if (tokens.size() < 3) {
+            std::cerr << "Add command requires KeyPath and Type" << std::endl;
+            return true;
+        } else {
+            std::queue<std::string> keyPath;
+            parseCommandKeyPathString(tokens[1], keyPath);
+            plist::ObjectType type = parseType(tokens[2]);
+            Set(root.object.get(), keyPath, type, parseCommandValueString(tokens.begin() + 3, tokens.end()), false);
+        }
+    } else if (command == "Clear") {
+        plist::ObjectType clearType;
+        if (tokens.size() < 2) {
+            clearType = plist::ObjectType::Dictionary;
+        } else {
+            clearType = parseType(tokens[1]);
+        }
+        Clear(root, clearType);
+    } else if (command == "Delete") {
+        if (tokens.size() < 2) {
+            std::cerr << "Add command requires KeyPath" << std::endl;
+            return true;
+        }
+        std::queue<std::string> keyPath;
+        parseCommandKeyPathString(tokens[1], keyPath);
+        Delete(root.object.get(), keyPath);
+    } else if (command == "Help") {
+        CommandHelp();
+    } else {
+        std::cerr << "Unrecognized command" << std::endl;
+    }
+    return true;
+}
+
+static const char *prompt(EditLine *e) {
+  return "Command: ";
+}
+
+int
+main(int argc, char **argv)
+{
+    std::vector<std::string> args = std::vector<std::string>(argv + 1, argv + argc);
+    auto filesystem = std::unique_ptr<libutil::Filesystem>(new libutil::DefaultFilesystem());
```

Comment:
nit: this can be just `libutil::DefaultFilesystem filesystem = libutil::DefaultFilesystem();` and skip the `std::unique_ptr`; see 11725f9b3a541ac97f568b679ad07719981f68b8